services/processor/tasks/genesis/genesis.go

```diff
@@ -154,28 +152,28 @@ func (p *ProcessGenesisSingletonTask) Task(job *work.Job) error {
 
 func (p *ProcessGenesisSingletonTask) storageMinerState(ctx context.Context, addr address.Address, act *types.Actor) (*genesismodel.GenesisMinerTaskResult, error) {
 	// actual miner actor state and miner info
-	var mstate miner.State
-	astb, err := p.node.ChainReadObj(ctx, act.Head)
+	mstate, err := miner.Load(p.node.Store(), act)
 	if err != nil {
 		return nil, err
 	}
-	if err := mstate.UnmarshalCBOR(bytes.NewReader(astb)); err != nil {
+
+	// miner raw and qual power
+	// TODO this needs caching so we don't re-fetch the power actors claim table for every tipset.
```

Comment:
Can you expand on this comment? This is the genesis block so I assumed we'd be running it just once

Reply:
Information about a miners power is contained in the power actors state. When we request a miners power lotus loads the power actors state (I suspect), extracts the requested miners power and returns it (maybe it got more efficient since I last checked). Since there are multiple miners contained in the genesis state we will likely be loading the power actors state multiple times, which is unnecessary. This is also true for loading a miners power in the miner task, and the same TODO exists there as well. I plan address these in a follow on if that's alright?

Reply:
Fine with me